### DISCUSSION TITLE
PDF PLUGIN o IMPLEMENTAZIONE

### DISCUSSION THREAD

**Discussion by tatobarresi** [rejetto/hfs#412] [Ideas]
Is there a way to open a PDF in browser view, and have the ability to edit it and automatically save it in the directory where the original was present?.

**Comment by rejetto** [comment]
editing pdf using just web is complex and beyond the scope of the HFS project, but of course one can consider making a plugin.

i wish browsers had a way to download any type of file in a sandbox, open the file with the local associated app for editing (asking user confirmation), and let the browser reupload after you save.
I made some research, and while saving and re-uploading from a sandbox seems to be possible with Chrome (not Firefox or others), it still lacks (apparently) the ability to open the associated local app.

**Comment by rejetto** [comment]
and yet this is what can be done, always limited to Chrome browser:
1. the user click on a file, in a folder with upload permission.
2. the user click a new menu new entry "Download and re-upload changes"
3. a dialog appears offering the user to save the file to the desktop (or maybe the download folder?). The user can decide to just save, or decide a different folder.
4. Now the user can open the file just downloaded on the desktop with its local editor app. Any time the file is saved, the file will be uploaded automatically to the server.

of course this implies that you have installed an editor app on your computer.
Is this scenario interesting? Any comments?

**Reply by tatobarresi** [reply]
Potrebbe essere seriamente la soluzione al problema. quindi anziché avere un editor dal browser, viene utilizzata l'app locale....non male.
Mi piacerebbe tanto provare questa nuova funzione.

**Comment by rejetto** [comment]
i've been working hard on it, and came up with a plugin.
Being it Chrome-only I don't think it should be included in HFS, so it will likely stay as a plugin.
It's published, but it requires some features of next release, so I'm giving you a preview so that you can test it
[hfs-windows.zip](https://github.com/rejetto/hfs/files/13694120/hfs-windows.zip)

the plugin name is "reupload"
<img width="132" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/c1e115db-9f2d-4a6e-81b2-11fa5ce90b79">

Let me know

**Comment by tatobarresi** [comment]
Buongiorno Massimo. Testerò il plugin oggi e ti farò sapere. grazie mille

**Comment by tatobarresi** [comment]
Hi Massimo, I can't find the menu "the user click a new menu new entry "Download and re-upload changes".  Am I doing something wrong?

**Comment by rejetto** [comment]
sorry for not mentioning it, the menu will be visible only if you have permission to overwrite files.
You need upload permission and disable "don't overwrite uploading" option. As an alternative to disabling the option, you can give delete permission.

**Comment by rejetto** [comment]
i've just found out that this only works with https 😭
